agent/src/main/java/com/cloud/agent/Agent.java

```diff
@@ -1228,7 +1228,14 @@ public void doTask(final Task task) throws TaskExecutionException {
                     logger.error("Error parsing task", e);
                 }
             } else if (task.getType() == Task.Type.DISCONNECT) {
-                logger.debug("Executing disconnect task - {}", () -> getLinkLog(task.getLink()));
+                try {
+                    // an issue has been found if reconnect immediately after disconnecting. please refer to https://github.com/apache/cloudstack/issues/8517
```

Comment:
```suggestion
                    // an issue has been found if reconnect immediately after disconnecting.
```

I think we can refer a github issue in a PR but not in code. We may not be on github forever.